### PR TITLE
Cards

### PR DESCRIPTION
**Pull request by StevenLocke**
TODO: Should replace `var`s with `let`s.

### CODE REVIEW

File: src/js/components/Card.js [hunk 1]
```diff
@@ -0,0 +1,41 @@
+var $ = require('jquery');
+
+import React from "react";
+
+import Event  from "../components/Event";
+
+export default class Card extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {type: props.type}
```
Comment (larajanecka) [inline src/js/components/Card.js]: missing semicolon

File: src/js/components/Card.js [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    switch (type) {
+      case "event":
+        renderedContent = <Event key={item.id} {...item}/>;
+        break;
+      default:
+        return null;
+    }
+
+    return (
+      <div class="Card">
```
Comment (larajanecka) [inline src/js/components/Card.js]: convention says classes start lowercase, don't bother if its annoying to change though, nbd

File: src/js/pages/Events.js [hunk 3]
```diff
@@ -33,7 +33,6 @@ export default class Events extends React.Component {
   }
 
 	getEvents() {
-    console.log(EventStore.events);
```
Comment (larajanecka) [inline src/js/pages/Events.js:36]: thanks for this one. mb

File: src/js/pages/Home.js [hunk 4]
```diff
@@ -3,105 +3,103 @@ var $ = require('jquery');
...
-      messageStack: this.state.messageStack.add({
-        message: message.message,
-        key: message.id,
-        dismissAfter: (message.expirytime - currTime) * 1000
-      })
-    });
-  }
+  componentDidMount() {
+    NotificationStore.getAll();
+    EventStore.getAll();
```
Comment (larajanecka) [inline src/js/pages/Home.js]: I wonder if any differences will occur with calling store loads in compnentDidMount vs componentWillMount

File: src/js/pages/Home.js [hunk 5]
```diff
@@ -3,105 +3,103 @@ var $ = require('jquery');
...
+    var unexpired = NotificationStore.getUnexpired();
+    let temp = [];
+    const currTime = moment().unix();
+    unexpired.forEach(function(notification) {
+      temp.push({
+        message: notification.attributes.message,
+        key: "notification-" + notification.attributes.id,
+        action: 'Dismiss',
+        dismissAfter: (notification.attributes.expirytime - currTime) * 1000,
+        onClick: () => this.removeNotification("notification-" + notification.attributes.id),
```
Comment (larajanecka) [inline src/js/pages/Home.js]: Do notifications not have id numbers that we should be filtering on? Its a bit weird to be intentionally building this string to check against.

File: src/js/pages/Home.js [hunk 6]
```diff
@@ -3,105 +3,103 @@ var $ = require('jquery');
...
 
-      this.messageStack = new OrderedSet();
-      this.state.messages.forEach(function(m) {
-        self.addNotification(m);
+  getNotifications() {
+    var unexpired = NotificationStore.getUnexpired();
+    let temp = [];
+    const currTime = moment().unix();
+    unexpired.forEach(function(notification) {
+      temp.push({
```
Comment (larajanecka) [inline src/js/pages/Home.js]: this should probably be `_.map(unexpired, (notification) => { ... }` if its a one to one

File: src/js/pages/Home.js [hunk 7]
```diff
@@ -3,105 +3,103 @@ var $ = require('jquery');
...
     });
 
     return (
       <div>
         <NotificationStack
-          notifications={this.state.messageStack.toArray()}
+          notifications={notifications}
           onDismiss={notification => this.setState({
-            notifications: this.state.messageStack.delete(notification)
+            notifications: notifications.filter((n) => n.key !== notification.key)
```
Comment (larajanecka) [inline src/js/pages/Home.js]: This should probably be a call to NotificationStore.dismiss  and like a refresh state.  We want stores to handle all the data record manipulations and the components to do the rendering.

**Review by larajanecka — COMMENTED** [review]
The notification dismiss logic should be moved to the notification store. Other than that it looks fairly good.

File: src/js/pages/Home.js (hunk 5 above)
Comment (StevenLocke) [reply src/js/pages/Home.js]: They do, the issue was that events may have the same id, so we build a type string to differentiate between them.

File: src/js/stores/NotificationStore.js [hunk 8]
```diff
@@ -0,0 +1,65 @@
+import { EventEmitter } from "events";
+
+import dispatcher from "../dispatcher";
+
+var $ = require("jquery");
+var moment = require('moment');
+var url = "https://dev.calligre.com"
```
Comment (TheKevJames) [inline src/js/stores/NotificationStore.js]: semicolon

**Review by larajanecka — APPROVED** [review]

**Pull request merged**